src/main.rs

```diff
@@ -340,3 +340,376 @@ fn format_duration(duration: &chrono::Duration) -> String {
 
     format!("{}d {}:{}:{}", days, hours, minutes, seconds)
 }
+
+#[cfg(test)]
```

Comment:
This is an integration test but I didn't want to do a lot of refactoring to make integration tests work. 
The reason the integration test requires refactoring is that we generate configs on-the-fly and as such we need access to `Config` structs. We generate these configs on-the-fly using string manipulation but for complex integration tests we will end up with a lot of mess.

Reply:
Why not write it in ruby?